src/main/java/com/fasterxml/jackson/databind/introspect/POJOPropertiesCollector.java

```diff
@@ -1450,6 +1450,50 @@ private PropertyNamingStrategy _findNamingStrategy()
                 return pns;
             }
         }
+
+        // PropertyNamingStrategy
+        if (namingClass == PropertyNamingStrategy.SnakeCaseStrategy.class) {
```

Comment:
can this list of `if` stmts be made into a `switch` stmt with cases instead?

Reply:
If we were to do this, I think switch logic should be moved into a static method of `PropertyNamingStrategies`, so here we would just call that method.

Reply:
Moved switching logic to `PropertyNamingStrategies` and added a test case.